test/e2e/storage/testsuites/provisioning.go

```diff
@@ -198,14 +208,34 @@ func testProvisioning(input *provisioningTestInput) {
 		input.pvc.Spec.VolumeMode = &block
 		TestDynamicProvisioning(input.testCase, input.cs, input.pvc, input.sc)
 	})
+
+	It("should provision storage with snapshot data source [Feature:VolumeSnapshotDataSource]", func() {
+		if !input.dInfo.Capabilities[CapDataSource] {
+			framework.Skipf("Driver %q does not support populate data from snapshot - skipping", input.dInfo.Name)
+		}
+		if input.dInfo.Name == "csi-hostpath-v0" {
```

Comment:
Can we configure the hostpath-v0 driver to not support the datsource capability and use that to skip?

Reply:
Both testdrivers are provided by https://github.com/kubernetes/kubernetes/blob/aa7220876a3129cac79deeb9a29a661155b66b2e/test/e2e/storage/drivers/csi.go#L57-L61

Yes, it would be possible to initialize that instance differently in https://github.com/kubernetes/kubernetes/blob/aa7220876a3129cac79deeb9a29a661155b66b2e/test/e2e/storage/drivers/csi.go#L248-L258.

I missed that this driver name check is in the provisioning testsuite. I agree that this is conceptually wrong.

Reply:
configure the hostpath-v0 driver to not support the datsource capability in the initialize. PTAL
```
	if name == "csi-hostpath-v0" {
		driver.driverInfo.Capabilities = map[testsuites.Capability]bool{
			testsuites.CapPersistence: true,
		}
	} else {
		driver.driverInfo.Capabilities = map[testsuites.Capability]bool{
			testsuites.CapPersistence: true,
			testsuites.CapDataSource:  true,
		}
	}
```